### DISCUSSION TITLE
Repeater & hasManyThrough?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#6564] [Help]
When the repeater was refactored to support relationships was support for hasManyThrough removed?  It seems like this was supported in the RelationshipRepeater.

```
Filament\Forms\Components\Repeater::getRelationship(): Return value must be of type Illuminate\Database\Eloquent\Relations\HasOneOrMany|Illuminate\Database\Eloquent\Relations\BelongsToMany|null, Illuminate\Database\Eloquent\Relations\HasManyThrough returned
```

**Comment by danharrin** [comment]
I can't remember supporting that, but you can PR if you like

**Reply by yaplej** [reply]
You are correct.  I was thinking of this post.  Just had to go re-read it.
https://github.com/filamentphp/filament/discussions/2645

I was able to get the functionality I wanted by using a very restricted relationship manager and then checking the state of each of those records in the relationship before allowing the record to move to a new state.

**Comment by classicalguss** [comment]
@danharrin I tried to make the getRelationship accept HasManyThrough and it worked without any error

`public function getRelationship(): HasOneOrMany | BelongsToMany | HasManyThrough | null`

Can I make a PR?

**Reply by danharrin** [reply]
See my comment on the other thread
